source/agora/consensus/protocol/Nominator.d

```diff
@@ -463,15 +474,17 @@ extern(D):
 
         Signs the SCPEnvelope with the node's private key.
 
-        todo: Currently not signing yet. To be done.
-
         Params:
             envelope = the SCPEnvelope to sign
 
     ***************************************************************************/
 
     public override void signEnvelope (ref SCPEnvelope envelope)
     {
+        const challenge = SCPStatementHash(&envelope.statement).hashFull();
+        const R = Pair.random();
+        const sig = sign(this.key_pair.v, this.key_pair.V, R.V, R.v, challenge);
```

Comment:
There's an overload that does the `random` call for you. `siggn(this.key_pair, challenge)` AFAIR.
And why the intermediate value ?

Reply:
Good catch, I'll update it.